SimPEG/Directives.py

```diff
@@ -328,25 +434,70 @@ def initialize(self):
         # Check if it is a ComboObjective
         if not isinstance(self.reg, Regularization.BaseComboRegularization):
 
-            # It is a Combo objective, so will have to loop
-            self.ComboObjFun = True
+            # It is a Combo objective, we will have to loop through a list
+            self.ComboRegFun = True
```

Comment:
@fourndo: instead of checking every time if the misfit and regularizations are combos, what about just casting the regularization and data misfit to always be combos (even if they are combos of 1). You could do this by adding
```python
@property
    def dmisfit(self):
        if getattr(self, '_dmisfit', None) is None:
            self.dmisfit = self.invProb.dmisfit  # go through the setter
        return self._dmisfit

    @dmisfit.setter
    def dmisfit(self, value):
        if not isinstance(value, ObjectiveFunction.ComboObjectiveFunction):
            value = 1*value  # turn it into a combo objective function
        self._dmisfit = value

    @property
    def reg(self):
        if getattr(self, '_reg', None) is None:
            self.reg = self.invProb.reg  # go through the setter
        return self._reg

    @reg.setter
    def reg(self, value):
        if isinstance(value, Regularization.BaseComboRegularization):
            value = 1*value  # turn it into a combo objective function
        self._reg = value
``` 

So regardless of if the data misfit or regularizations are combos or not, they are just cast to a combo and you always loop. 

![image](https://cloud.githubusercontent.com/assets/6361812/25072656/1833b31c-2289-11e7-9643-32b4eac33233.png)